### PR TITLE
[TACHYON-1428] Documented parts of the tachyon-shell module in preparation for the checkstyle rule

### PR DESCRIPTION
**Pull request by HorizonNet**
https://tachyon.atlassian.net/browse/TACHYON-1428

### CODE REVIEW

File: shell/src/main/java/tachyon/shell/command/TouchCommand.java [hunk 1]
```diff
@@ -29,6 +29,12 @@
  */
 public final class TouchCommand extends AbstractTfsShellCommand {
 
+  /**
+   * Constructs a new instance to create a 0 byte file specified by argv.
```
Comment (aaudiber) [inline shell/src/main/java/tachyon/shell/command/TouchCommand.java]: no need to add "specified by argv"

File: shell/src/main/java/tachyon/shell/TfsShellUtils.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ public static String validatePath(String path, TachyonConf tachyonConf) throws I
    * @param tachyonClient the client used to fetch information of Tachyon files
    * @param inputURI the input URI (could contain wildcards)
    * @return a list of {@link TachyonURI}s that matches the inputURI
-   * @throws IOException
+   * @throws IOException if the client cannot open the directory or list its status
```
Comment (uronce-cc) [inline shell/src/main/java/tachyon/shell/TfsShellUtils.java]: It's a bit confusing, does `the directory` mean `inputURI`? Do you feel that `if any filesystem errors are encountered when expanding paths with wildcards` describes the intention?

File: shell/src/main/java/tachyon/shell/TfsShellUtils.java [hunk 3]
```diff
@@ -120,7 +120,7 @@ public static String validatePath(String path, TachyonConf tachyonConf) throws I
    * @param parentDir the {@link TachyonURI} of the directory in which we are searching matched
    *                  files
    * @return a list of {@link TachyonURI}s of the files that match the inputURI in parentDir
-   * @throws IOException
+   * @throws IOException if the client cannot open the directory or list its status
```
Comment (uronce-cc) [inline shell/src/main/java/tachyon/shell/TfsShellUtils.java]: ditto

File: shell/src/main/java/tachyon/shell/command/TouchCommand.java (hunk 1 above)
Comment (uronce-cc) [inline shell/src/main/java/tachyon/shell/command/TouchCommand.java]: I feel like there is no need to duplicate the class's purpose in the constructor comment, it's enough to just put it in class comment.

**Comment by HorizonNet** [comment]
@aaudiber Thanks for the comments. @uronce-cc I took your description for the exception. Seems to be a better fit. I also removed the duplication of the class comment in the command constructors. I will create another PR to make it consistent in the package.

File: shell/src/main/java/tachyon/shell/TfsShellUtils.java [hunk 4]
```diff
@@ -205,12 +209,13 @@ public static String validatePath(String path, TachyonConf tachyonConf) throws I
...
 
   /**
-   * The characters that have special regex semantics
+   * The characters that have special regex semantics.
    */
   private static final Pattern SPECIAL_REGEX_CHARS = Pattern.compile("[{}()\\[\\].+*?^$\\\\|]");
 
   /**
-   * Escape the special characters in a given string
+   * Escape the special characters in a given string.
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/TfsShellUtils.java]: Escape -> Escapes

File: shell/src/main/java/tachyon/shell/TfsShellUtils.java [hunk 5]
```diff
@@ -219,14 +224,15 @@ private static String escape(String str) {
   }
 
   /**
-   * Replace the wildcards with Java's regex semantics
+   * Replace the wildcards with Java's regex semantics.
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/TfsShellUtils.java]: Replace -> Replaces

File: shell/src/main/java/tachyon/shell/TfsShellUtils.java [hunk 6]
```diff
@@ -219,14 +224,15 @@ private static String escape(String str) {
...
    */
   private static String replaceWildcards(String text) {
     return escape(text).replace("\\*", ".*");
   }
 
   /**
-   * Return whether or not fileURI matches the patternURI
+   * Return whether or not fileURI matches the patternURI.
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/TfsShellUtils.java]: Return -> Returns

File: shell/src/main/java/tachyon/shell/TfsShellUtils.java [hunk 7]
```diff
@@ -236,7 +242,8 @@ private static boolean match(TachyonURI fileURI, TachyonURI patternURI) {
   }
 
   /**
-   * Return whether or not filePath matches patternPath
+   * Return whether or not filePath matches patternPath.
```
Comment (yupeng9) [inline shell/src/main/java/tachyon/shell/TfsShellUtils.java]: Return -> Returns

**Comment by yupeng9** [comment]
Thanks for the doc improvement. I left some comments.

**Comment by HorizonNet** [comment]
@yupeng9 Thanks for the comments. I updated the PR accordingly.

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**